com.unity.ml-agents/Runtime/Sensors/CompressionSpec.cs

```diff
@@ -0,0 +1,112 @@
+    /// A description of the compression used for observations.
+    /// </summary>
+    /// <remarks>
+    /// Most ISensor implementations can't take advantage of compression,
+    /// and should return CompressionSpec.Default() from their ISensor.GetCompressionSpec() methods.
+    /// Visual observations, or mulitdimensional categorical observations (for example, image segmentation
+    /// or the piece types in a match-3 game board) can use PNG compression reduce the amount of
+    /// data transferred between Unity and the trainer.
+    /// </remarks>
+    public struct CompressionSpec
+    {
+        internal SensorCompressionType m_SensorCompressionType;
+
+        /// <summary>
+        /// The compression type that the sensor will use for its observations.
+        /// </summary>
+        public SensorCompressionType SensorCompressionType
+        {
+            get => m_SensorCompressionType;
+        }
+
+        internal int[] m_CompressedChannelMapping;
+
+        /// The mapping of the channels in compressed data to the actual channel after decompression.
+        /// The mapping is a list of integer index with the same length as
+        /// the number of output observation layers (channels), including padding if there's any.
+        /// Each index indicates the actual channel the layer will go into.
+        /// Layers with the same index will be averaged, and layers with negative index will be dropped.
+        /// For example, mapping for CameraSensor using grayscale and stacking of two: [0, 0, 0, 1, 1, 1]
+        /// Mapping for GridSensor of 4 channels and stacking of two: [0, 1, 2, 3, -1, -1, 4, 5, 6, 7, -1, -1]
+        public int[] CompressedChannelMapping
+        {
+            get => m_CompressedChannelMapping;
+        }
+
+        /// <summary>
+        /// Return a CompressionSpec indicating no compression. This is recommended for most sensors.
+        /// </summary>
+        /// <returns></returns>
+        public static CompressionSpec Default()
+        {
+            return new CompressionSpec
+            {
+                m_SensorCompressionType = SensorCompressionType.None,
+                m_CompressedChannelMapping = null
+            };
+        }
+
+        /// <summary>
+        /// Return a CompressionSpec indicating possible compression.
+        /// </summary>
+        /// <param name="sensorCompressionType">The compression type to use.</param>
+        /// <param name="compressedChannelMapping">Optional mapping mapping of the channels in compressed data to the actual channel after decompression.</param>
+        /// <returns></returns>
+        public static CompressionSpec Compressed(SensorCompressionType sensorCompressionType, int[] compressedChannelMapping = null)
+        {
+            return new CompressionSpec
+            {
+                m_SensorCompressionType = sensorCompressionType,
+                m_CompressedChannelMapping = compressedChannelMapping
+            };
```

Comment:
technically you can pass in SensorCompressionType.None and non-null mapping?

Reply:
Yeah, you can. I think I'll just make this the constructor instead of using a static method.